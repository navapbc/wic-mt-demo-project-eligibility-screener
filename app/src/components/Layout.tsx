import { Trans } from 'next-i18next'
import Image from 'next/image'
import { ReactElement } from 'react'

type Props = {
  children: ReactElement
}

const Layout = ({ children }: Props): ReactElement => {
  return (
    <div className="container">
      <header className="header usa-header usa-header--basic" role="banner">
        <div className="usa-navbar">
          <div className="grid-row">
            <div className="desktop:grid-col-8">
              <div className="usa-logo margin-left-2">
                <em className="usa-logo__text">
                  <Trans i18nKey="Layout.header" />
                </em>
              </div>
            </div>
          </div>
        </div>
      </header>
      <main className="main">
        <div className="grid-row">
          <div className="desktop:grid-col-8 padding-2 padding-bottom-8">
            {children}
          </div>
        </div>
      </main>
      <footer className="footer usa-footer usa-footer--slim">
        <div className="usa-footer__primary-section">
          <div className="grid-row">
            <div className="desktop:grid-col-8 padding-2">
              <div className="logos">
                <Image
                  src="/img/wic-logo.svg"
                  alt="WIC logo"
                  width={64.52}
                  height={32}
                />
                <Image
                  src="/img/montana-logo.svg"
                  alt="Monthana DPHHS logo"
                  width={46.22}
                  height={32}
                />
              </div>
              <div className="font-body-3xs">
                <p>
                  <Trans
                    components={[
                      <a
                        key="0"
                        href="https://dphhs.mt.gov/ecfsd/wic/index"
                        className="usa-link usa-link--external"
                        target="_blank"
                        rel="noopener noreferrer"
                      />,
                      <a
                        key="1"
                        href="https://www.signupwic.com/"
                        className="usa-link usa-link--external"
                        target="_blank"
                        rel="noopener noreferrer"
                      />,
                    ]}
                    i18nKey={'Layout.footer1'}
                  />
                </p>
                <p>
                  <Trans
                    components={[
                      <a
                        key="0"
                        href="https://www.fns.usda.gov/civil-rights/usda-nondiscrimination-statement-other-fns-programs"
                        className="usa-link usa-link--external"
                        target="_blank"
                        rel="noopener noreferrer"
                      />,
                    ]}
                    i18nKey={'Layout.footer2'}
                  />
                </p>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  )
}

export default Layout
