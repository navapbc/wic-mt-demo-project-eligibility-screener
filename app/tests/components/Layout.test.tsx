// test/pages/index.test.js
import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'

import Layout from '@components/Layout'

describe('Layout', () => {
  it('should render placeholder header text', () => {
    render(<Layout children={<h1>'child'</h1>} />)

    const header = screen.getByText(/WIC Eligibility Screener/i)

    expect(header).toBeInTheDocument()
    expect(header).toMatchSnapshot()
  })

  it('should pass accessibility scan', async () => {
    const { container } = render(<Layout children={<h1>'child'</h1>} />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
