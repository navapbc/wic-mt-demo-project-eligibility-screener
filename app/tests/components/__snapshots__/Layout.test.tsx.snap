// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header usa-header usa-header--basic"
      role="banner"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="grid-row"
        >
          <div
            class="desktop:grid-col-8"
          >
            <div
              class="usa-logo margin-left-2"
            >
              <em
                class="usa-logo__text"
              >
                Apply for WIC in Montana
              </em>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="grid-row"
      >
        <div
          class="desktop:grid-col-8 padding-2 padding-bottom-8"
        >
          <h1>
            'child'
          </h1>
        </div>
      </div>
    </main>
    <footer
      class="footer usa-footer usa-footer--slim"
    >
      <div
        class="usa-footer__primary-section"
      >
        <div
          class="grid-row"
        >
          <div
            class="desktop:grid-col-8 padding-2"
          >
            <div
              class="logos"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2764.52%27%20height=%2732%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="WIC logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2746.22%27%20height=%2732%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Monthana DPHHS logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="font-body-3xs"
            >
              <p>
                This site is a demonstration project built with 
                <a
                  class="usa-link usa-link--external"
                  href="https://dphhs.mt.gov/ecfsd/wic/index"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Montana WIC
                </a>
                 and is for example purposes only. If you need to submit an application for WIC, please use 
                <a
                  class="usa-link usa-link--external"
                  href="https://www.signupwic.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  SignUpWIC.com
                </a>
                 to find a location near you.
              </p>
              <p>
                This institution is an equal opportunity provider. 
                <a
                  class="usa-link usa-link--external"
                  href="https://www.fns.usda.gov/civil-rights/usda-nondiscrimination-statement-other-fns-programs"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  USDA Non-Discrimination Statement
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
